server/src/services/library.service.ts

```diff
@@ -465,62 +443,154 @@ export class LibraryService extends BaseService {
+  async handleSyncAssets(job: JobOf<JobName.LIBRARY_SYNC_ASSETS>): Promise<JobStatus> {
+    const assets = await this.assetRepository.getByIds(job.assetIds);
+
+    const assetIdsToOffline: string[] = [];
+    const assetIdsToOnline: string[] = [];
+    const assetIdsToUpdate: string[] = [];
+
+    this.logger.debug(`Checking batch of ${assets.length} existing asset(s) in library ${job.libraryId}`);
+
+    for (const asset of assets) {
+      const action = await this.checkExistingAsset(asset, job.libraryId);
+      switch (action) {
+        case AssetSyncResult.OFFLINE: {
+          assetIdsToOffline.push(asset.id);
+          break;
+        }
+        case AssetSyncResult.UPDATE: {
+          assetIdsToUpdate.push(asset.id);
+          break;
+        }
+        case AssetSyncResult.CHECK_OFFLINE: {
+          const isInImportPath = job.importPaths.find((path) => asset.originalPath.startsWith(path));
+
+          if (!isInImportPath) {
+            this.logger.verbose(
+              `Offline asset ${asset.originalPath} is still not in any import path, keeping offline in library ${job.libraryId}`,
+            );
+            break;
+          }
 
-    const markOffline = async (explanation: string) => {
-      if (!asset.isOffline) {
-        this.logger.debug(`${explanation}, removing: ${asset.originalPath}`);
-        await this.assetRepository.updateAll([asset.id], { isOffline: true, deletedAt: new Date() });
+          const isExcluded = job.exclusionPatterns.some((pattern) => picomatch.isMatch(asset.originalPath, pattern));
+
+          if (!isExcluded) {
+            this.logger.debug(`Offline asset ${asset.originalPath} is now online in library ${job.libraryId}`);
+            assetIdsToOnline.push(asset.id);
+            break;
+          }
+
+          this.logger.verbose(
+            `Offline asset ${asset.originalPath} is in an import path but still covered by exclusion pattern, keeping offline in library ${job.libraryId}`,
+          );
+
+          break;
+        }
       }
-    };
+    }
 
-    const isInPath = job.importPaths.find((path) => asset.originalPath.startsWith(path));
-    if (!isInPath) {
-      await markOffline('Asset is no longer in an import path');
-      return JobStatus.SUCCESS;
+    const progressMessage: string[] = [];
+
+    if (assetIdsToOffline.length > 0) {
+      await this.assetRepository.updateAll(assetIdsToOffline, {
+        isOffline: true,
+        status: AssetStatus.TRASHED,
+        deletedAt: new Date(),
+      });
+
+      progressMessage.push(`${assetIdsToOffline.length} offlined`);
     }
 
-    const isExcluded = job.exclusionPatterns.some((pattern) => picomatch.isMatch(asset.originalPath, pattern));
-    if (isExcluded) {
-      await markOffline('Asset is covered by an exclusion pattern');
-      return JobStatus.SUCCESS;
+    if (assetIdsToOnline.length > 0) {
+      //TODO: When we have asset status, we need to leave deletedAt as is when status is trashed
```

Comment:
This TODO does not need to be addressed here?

Reply:
Fixed